main.go

```diff
@@ -272,3 +287,37 @@ func (m *metricHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		closer.Close()
 	}
 }
+
+func detectSharding(kubeClient clientset.Interface, podName, namespaceName string) (shard, totalShards uint64, err error) {
+	glog.Infof("Auto detecting shard.")
+	p, err := kubeClient.CoreV1().Pods(namespaceName).Get(podName, metav1.GetOptions{})
+	if err != nil {
+		return 0, 0, err
+	}
+
+	owners := p.GetOwnerReferences()
+	for _, o := range owners {
+		if o.APIVersion == "apps/v1" && o.Kind == "StatefulSet" {
```

Comment:
wether we need to detect pod actual Controller?

Reply:
Not strictly necessary I think, it's a heuristic anyways, as theoretically there could be any number of statefulsets in the ownerreferences, and we just pick the first occurrence. I'll add it anyways though.